src/Resources/ResourceFactory.php

```diff
@@ -80,7 +80,7 @@ public function getModelClass($model, $collection = false)
             if (count($model) == 0) {
                 return ResourceModel::class;
             } else {
-                $model = $model[0];
+                $model = $model[0] ?? $model->first();
```

Comment:
I get why this happened. The collection could have ids as keys.

Shorter version?
```suggestion
                $model = Arr::first($model);
```